src/cpp/xmlparser/XMLDynamicParser.cpp

```diff
@@ -463,7 +463,14 @@ XMLP_ret XMLParser::parseXMLAliasDynamicType(
                     alias_descriptor->base_type(value_type);
                     DynamicTypeBuilder::_ref_type builder {DynamicTypeBuilderFactory::get_instance()->create_type(
                                                                alias_descriptor)};
-                    if (false == XMLProfileManager::insertDynamicTypeByName(name, builder->build()))
+                    if (nullptr != builder)
+                    {
+                        if (false == XMLProfileManager::insertDynamicTypeByName(name, builder->build()))
```

Comment:
We may just group into one clause

```suggestion
                        if (nullptr != builder && false == XMLProfileManager::insertDynamicTypeByName(name, builder->build()))
```